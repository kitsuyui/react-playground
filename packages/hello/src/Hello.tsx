import type React from 'react'

/**
 * A simple Hello World component for React.
 * @returns A JSX element that displays "Hello, world!".
 */
export const Hello = (): React.JSX.Element => {
  return <span>Hello, world!</span>
}
