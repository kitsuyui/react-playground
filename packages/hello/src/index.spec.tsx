import { describe, expect, it } from 'vitest'

import { render } from '@testing-library/react'

import * as module from './index'
import { Hello } from './Hello'

describe('exports', () => {
  it('should export Hello component', () => {
    expect(module.Hello).toBeDefined()
    expect(module.Hello).toBeTypeOf('function')
  })
})

describe('Hello component', () => {
  it('should render properly', () => {
    const { asFragment } = render(<Hello />)
    expect(asFragment()).toMatchSnapshot()
    expect(asFragment().querySelector('span')).toBeDefined()
    expect(asFragment().querySelector('span')?.textContent).toBe('Hello, world!')
  })
})
