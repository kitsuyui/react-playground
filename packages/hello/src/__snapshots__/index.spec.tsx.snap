// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hello component > should render properly 1`] = `
<DocumentFragment>
  <span>
    Hello, world!
  </span>
</DocumentFragment>
`;
